fix: add missing cover image for password-generator

saatvik-10 opened the pull request:
## Problem
The Password Generator project was displaying a broken image icon in the Trending Plays section.

## Root Cause
- External image URL: `https://securityintelligence.com/wp-content/uploads/2018/10/si-eight-character-password-feature.jpg`
- This URL redirects to IBM security page and is no longer accessible as an image

## Solution
- Added local `cover.png` as fallback following existing pattern
- The `loadCoverImage` utility will now find the local file when external URL fails

## Testing
- Verified cover.png exists and is valid PNG format
- Follows same structure as other plays (analog-clock, etc.)

priyankarpal reviewed (changes requested):
still it's visible
<img width="378" height="406" alt="Image" src="https://github.com/user-attachments/assets/105e2494-46a9-417e-8ccd-53980b8d299d" />

saatvik-10 commented:
> still it's visible <img alt="Image" width="378" height="406" src="https://private-user-images.githubusercontent.com/88102392/466378360-105e2494-46a9-417e-8ccd-53980b8d299d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI1Njc5NDMsIm5iZiI6MTc1MjU2NzY0MywicGF0aCI6Ii84ODEwMjM5Mi80NjYzNzgzNjAtMTA1ZTI0OTQtNDZhOS00MTdlLThjY2QtNTM5ODBiOGQyOTlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE1VDA4MjA0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJiZWVlYTk5MTZlZWFjM2EwMDMwNDUyMDliMTA1MTY4YmI3NDE4NmEwYTkwZTk1MDk4Y2ZkYWRiNTYzOTE3NWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.OB8HMgDnbiT1nqnVvgMQynTzx3ZGvM8pE4XXm2Bhgqk">

Enhanced Image Loading Logic
- Modified `PlayThumbnail.jsx` to properly handle external image loading failures
- Before: `if (play.cover) use external, else use local`
- After: `try external image → if fails → fallback to local image`
- Added proper error handling with Image() constructor to detect load failures
